App now gets display names from display EDID

Soggy-Pancake opened the pull request:
The app now sets the correct connection for xrandr in self.displays. self.displays now is a nested array contaning the connection and the monitor name (ex: ["dp-1", "monitor"]), which should solve #220. 

I also fixed the imports as it would crash upon trying to import anything under brightness_controler_linux as the init.py had been moved to the same folder some time back.

Commit:
App correctly gets display names from display EDID and
sets the correct connection for xrandr in self.displays.
self.displays now is a nested array contaning the connection
and the monitor name (ex: ["dp-1", "monitor"])

Commit:
Remove extra print statement

LordAmit commented:
Hey! Thanks for the pull request and for addressing #220 ! 

Right now, I do not have a Linux machine to test this. @archisman-panigrahi , can you please help me by taking a look at it? Since the import statements changed a bit, your scripts to build packages might break, so I want your attention on this part!

Commit:
Added fallback to display connection name  if
monitor name extraction fails

Soggy-Pancake commented:
I just added a protection for when the display doesnt give EDID or the name cant be found for some reason. It will fall back to the connection name (ex. DP-0) if the name extraction fails for any reason.

archisman-panigrahi commented:
Sorry about the delay responding. I will review it this weekend. 

> your scripts to build packages might break, so I want your attention on this part!

The ddcutil version is not in PPA because I am yet to figure out how to package this new version.

Commit:
Add verbosity arguements for debugging without shoving things in console by default

Commit:
Updated display validity check in init to match new display array setup ["connection name", "display name"]

Commit:
Display reordering is now directly after discovering displays to prevent combo boxes being filled with the original xrandr screen order. I also fixed the display update to respect combo box selections as well as updating display sliders after changing either combo box.

Commit:
Fixed laptop brightness slider not being reenabled after leaving ddcutil mode.

Commit:
Secondary Brightness slider enabled after leaving ddcutil mode

Commit:
Fixed crashing on Wayland! Monitor names are missing now since xrandr doesnt give edid anymore.

Commit:
The old software brightness setting is broken on wayland, but display names are now gotten on wayland! DDC control now functions on both x11 and wayland.

Soggy-Pancake commented:
Wayland crashes have been fixed as I just managed to get wayland working yesterday. The original software brightness doesn't work on Wayland but ddc now will.

Commit:
restored original poetry imports.

Commit:
Added warning message for wayland users. I should probably
make this a ui popup instead of only in the terminal,
but at least there will be a more definitive way of telling
if users are on wayland.

Commit:
Added checks to prevent laptop displays from being set. The sliders will be disabled if it is a laptop display and there is another check in the setBrightness function as well just in case. The subprocess.run is now also wrapped in a try block as a last resort.

Commit:
added try block to xrandr execution as well.

Commit:
Now forcing ddcutil on wayland.

Commit:
Fix pyproject using lowercase readme.

Commit:
allow PyQt5 to be imported instead if qtpy fails to import.

Commit:
Undo pyqt5 import

Commit:
Created logging functions.

Commit:
Added logging EVERYWHERE

Soggy-Pancake committed:
Add extra check for user missing i2c group and alert user with instructions to fix.

Soggy-Pancake committed:
Add extra check for user missing i2c group and alert user with instructions to fix.

Soggy-Pancake committed:
Merge branch 'master' of https://github.com/Soggy-Pancake/Brightness-Direct-Control
